IAM / Cleanup / SDK

tanmayb123 opened the pull request:
1. Fixed IAM
1. Cleaned up / formatted code
1. Added SDK (from manual REST)

stevemart commented on Sources/HackernewsNLU/Controller.swift:
```diff
@@ -189,75 +175,71 @@ public class Controller {
-            try theData = theBody.rawData(options: JSONSerialization.WritingOptions())
+    func callNLUApi(newsArticle article: NewsArticles,completion: @escaping (SwiftyJSON.JSON)->(), failure: @escaping (String) -> Void) -> Void {
+        let service: NaturalLanguageUnderstanding!
+        if nluCreds["apikey"] == nil {
```
`nluCreds` is set by calling `initService`. `initService` doesn't seem to handle the IAM creds. I think that `initService` needs to be updated for this if statement to ever go into the else block.

stevemart commented on Sources/HackernewsNLU/Controller.swift:
```diff
@@ -189,78 +176,84 @@ public class Controller {
+        let apikeyValid = checkValidity(apikey: nluCreds["apikey"])
+        let userpassValid = checkValidity(username: nluCreds["username"], password: nluCreds["password"])
+        if apikeyValid && userpassValid {
+            failure("You provided both a username and password, as! well as! an API key. Please provide only one.")
```
why the exclamation points? `as! well as! an`

tanmayb123 replied:
will fix

tanmayb123 commented:
Total list of changes:
1. Added SDK support
1. Added IAM support
1. Enforced thread safety
1. Linted code (indentation, functions, renamed classes)
1. Optimized startup time (removed initial download, will now download when requested)
1. Minimized code for "NewsArticle" comparisons

tanmayb123 commented:
I'm going to implement @sanjeevghimire's feedback in another PR. I want to get this one merged in first. Is that alright @stevemart?
